Emit enums as proper Closure types.

mprobst opened the pull request:
This was previously avoided to keep the output valid TypeScript wise, but that
is not necessary, as we do not type check the intermediate output.

This has a remaining issue where it will emit Closure types for const enums that
disappear in runtime emit. We will need to investigate whether we should emit
the enum type for those.

Fixes #535.

mprobst commented:
Needs a g3 run to confirm working.

evmar commented on src/tsickle.ts:
```diff
@@ -1541,31 +1535,34 @@ class Annotator extends ClosureRewriter {
       const value = members.get(member)!;
+      this.emit(`${member}: `);
       if (typeof value === 'number') {
         this.emit(value.toString());
```
I wonder why this branch is necessary, it seems you could just always visit value

mprobst replied:
The number is actually a literal number, not a ts Node.

mprobst commented:
PTAL. I haven't cleaned up the CL descriptions, but essentially I changed this to always split the enum declaration into enum and export (because otherwise local type references don't work), and changed it to always emit a type, even in untyped mode, as that's harmless and essentially what we do already.

This requires tsickle to not run with `.d.ts` emit.
